packages/dd-trace/src/appsec/reporter.js

```diff
@@ -140,14 +140,14 @@ function filterExtendedHeaders (headers, excludedHeaderNames, tagPrefix, limit =
   return result
 }
 
-function getCollectedHeaders (req, res, shouldCollectEventHeaders) {
+function getCollectedHeaders (req, res, shouldCollectEventHeaders, storedResponseHeaders) {
   // Mandatory
   const mandatoryCollectedHeaders = filterHeaders(req.headers, REQUEST_HEADERS_MAP)
 
   // Basic collection
   if (!shouldCollectEventHeaders) return mandatoryCollectedHeaders
 
-  const responseHeaders = res.getHeaders()
+  const responseHeaders = storedResponseHeaders || res.getHeaders()
```

Comment:
shouldn't this be a merge of these things? whatever is in `storedResponseHeaders` + `res.getHeaders()`? Like:
```javascript
const responseHeaders = { ...storedResponseHeaders, ...res.getHeaders() }
```

Reply:
Yes, in fastify for example, if for some reason (a lib or whatever) is using `setHeaders` in `reply.raw` we will have different data in `storedResponseHeaders` and in `res.getHeaders`.

Corner cases, but they are cases anyway.